pinata/pinata.go

```diff
@@ -0,0 +1,225 @@
+package pinata
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/opensaucerer/bifrost/shared/errors"
+	"github.com/opensaucerer/bifrost/shared/types"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+	 "encoding/json"
+)
+
+type MetaData struct {
+    Name   string      `json:"name"`
+}
+
+type SuccessResponse struct {
+	 IpfsHash   string `json:"IpfsHash"`
+	 Timestamp 	string `json:"Timestamp"`
+	 PinSize 		int64 `json:"PinSize"`
+}
+
+/*
+	Upload File to pinata IPFS
+*/
+
+func (p PinataIPFSStorage) UploadFile(path, filename string, options map[string]interface{}) (*types.UploadedFile, error) {
+
+  // validate JWT	
+	success,err := p.TestAuthentication()
+	if err != nil {
+		return nil, &errors.BifrostError{
+			Err:       fmt.Errorf("problem authenticating: %v", err),
+			ErrorCode: errors.ErrBadRequest,
+		}
+	}
+	if !success {
+		return nil, &errors.BifrostError{
+			Err:       fmt.Errorf("authentication failed"),
+			ErrorCode: errors.ErrBadRequest,
+		}
+	}
+
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return nil, &errors.BifrostError{
+			Err:       fmt.Errorf("file does not exist: %s", path),
+			ErrorCode: errors.ErrBadRequest,
+		}
+	}
+
+	url := "https://api.pinata.cloud/pinning/pinFileToIPFS"
```

Comment:
This is an internal configuration and should be located inside the shared config package. 

Inside `shared/config` create a new file `url.go` and add it as
```go
package config

// URL constants.
const (
    // PinataPinFile is the API endpoint for pinning files and folders to pinata cloud
    PinataPinFile = "url-goes-here"
    // add comment
    PinataPinJSON = "url-goes-here"
    // add comment
    PinataPinCID = "url-goes-here"
)
```